test/suite/test_chunkcache03.py

```diff
@@ -0,0 +1,137 @@
+        else:
+            chunk_cache_extra_config = 'type=FILE,storage_path=/tmp/chunk_cache_{}'.format(randrange(0, 1000000000))
+
+        return 'tiered_storage=(auth_token=Secret,bucket=bucket2,bucket_prefix=pfx_,name=dir_store),' \
+            'chunk_cache=[enabled=true,chunk_size=512KB,capacity=20GB,pinned=("table:chunkcache01", "table:chunkcache02"),{}],'.format(chunk_cache_extra_config)
+
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('storage_sources', 'dir_store')
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+    
+    def insert(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor[ds.key(i)] = str(i) * 100
+
+    def read_and_verify(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor.set_key(ds.key(i))
+            cursor.search()
+            self.assertEqual(cursor.get_value(), str(i) * 100)
+
+    def test_chunkcache03(self):
+        uris = ["table:chunkcache01", "table:chunkcache02", "table:chunkcache03", "table:chunkcache04"]
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in uris]
+            
+        # Insert data in four tables.
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+        
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+        
+        # Reopen wiredtiger to migrate all data to disk.
+        self.reopen_conn()
+
+        # Read from the unpinned URIs and capture chunks in use.
+        self.read_and_verify(uris[2], ds[2])
+        self.read_and_verify(uris[3], ds[3])
+        chunks_inuse_excluding_pinned = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_inuse)
+        self.assertGreater(chunks_inuse_excluding_pinned, 0)
+
+        # Assert none of the chunks are pinned.
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned), 0)
+
+        # Read from the pinned URIs.
+        self.read_and_verify(uris[0], ds[0])
+        self.read_and_verify(uris[1], ds[1])
+        chunks_inuse_including_pinned = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_inuse)
+        self.assertGreater(chunks_inuse_including_pinned, chunks_inuse_excluding_pinned)
+
+        # Assert that chunks are pinned.
+        pinned_chunks_inuse = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned)
+        self.assertGreater(pinned_chunks_inuse, 0)
+
+        # Assert that the difference b/w the total chunks present and the unpinned chunks equal pinned chunks.
+        # This proves that the chunks read form pinned objects were all pinned. 
```

Comment:
Nitpick: form -> from.

Reply:
Updated in [c2195b0](https://github.com/wiredtiger/wiredtiger/pull/9555/commits/c2195b03581ef2cd05b1229fde134027b54fc589)